server/server/src/metrics.rs

```diff
@@ -235,28 +233,45 @@ pub fn get_user_info(
 }
 
 fn get_bytes_and_documents_count(
-    db: &mut ServerDb, owner: Owner, ids: Vec<Uuid>,
+    db: &mut ServerDb, owner: Owner,
 ) -> Result<(i64, u64), ServerError<MetricsError>> {
-    let mut total_documents = 0;
-    let mut total_bytes = 0;
-
-    for id in ids {
-        let metadata = db.metas.data().get(&id).ok_or_else(|| {
-            internal!("Could not get file metadata during metrics for {:?}", owner)
-        })?;
-
-        if metadata.is_document() {
-            if metadata.document_hmac().is_some() {
-                let usage = db.sizes.data().get(&id).ok_or_else(|| {
-                    internal!("Could not get file usage during metrics for {:?}", owner)
-                })?;
+    let total_bytes = get_usage_helper(db, &owner.0)
+        .unwrap()
+        .iter()
+        .map(|f| f.size_bytes)
+        .sum();
+    let total_documents = db.owned_files.data().get(&owner).unwrap().len() as i64;
 
-                total_bytes += usage;
-            }
+    Ok((total_documents, total_bytes))
+}
 
-            total_documents += 1;
-        }
-    }
+fn is_user_active<T: lockbook_shared::tree_like::TreeLike<F = ServerFile>>(
+    tree: LazyTree<T>, ids: &[Uuid],
+) -> Result<bool, ServerError<MetricsError>> {
+    let root = ids
+        .iter()
+        .find(|id| {
+            let file = tree.find(id).unwrap();
+            file.parent() == file.id()
+        })
+        .unwrap();
+    let root_creation_timestamp = tree.find(root)?.clone().file.timestamped_value.timestamp;
 
-    Ok((total_documents, total_bytes))
+    let latest_upsert = tree
+        .all_files()?
+        .iter()
+        .max_by(|x, y| {
+            x.file
+                .timestamped_value
+                .timestamp
+                .cmp(&y.file.timestamped_value.timestamp)
+        })
+        .unwrap()
+        .file
+        .timestamped_value
+        .timestamp;
```

Comment:
I think `last_seen` is still the right thing to use, not the most recent upsert.